ios/Exponent/Kernel/Views/EXAppViewController.m

```diff
@@ -315,6 +318,32 @@ - (void)errorViewDidSelectRetry:(EXErrorView *)errorView
 
 - (UIInterfaceOrientationMask)supportedInterfaceOrientations
 {
+  NSString *validatedVersion = [[EXVersions sharedInstance] availableSdkVersionForManifest:_appRecord.appLoader.manifest];
+  if ([@[@"33.0.0", @"34.0.0", @"35.0.0"] containsObject:validatedVersion]) {
+    return [self legacySupportedOrientationsLogic];
+  }
+  
+  if (_appRecord.experienceId == nil) {
+    return UIInterfaceOrientationMaskAllButUpsideDown;
+  }
+  EXScreenOrientationRegistry *registry = (EXScreenOrientationRegistry *)[UMModuleRegistryProvider getSingletonModuleForClass:[EXScreenOrientationRegistry class]];
+  if ([registry doesKeyExistForAppId:_appRecord.experienceId]) {
+    return [registry getOrientationMaskForAppId:_appRecord.experienceId];
+  } else if (_appRecord.appLoader.manifest) {
+    NSString *orientationConfig = _appRecord.appLoader.manifest[@"orientation"];
+    if ([orientationConfig isEqualToString:@"portrait"]) {
+      // lock to portrait
+      return UIInterfaceOrientationMaskPortrait;
+    } else if ([orientationConfig isEqualToString:@"landscape"]) {
+      // lock to landscape
+      return UIInterfaceOrientationMaskLandscape;
+    }
+  }
+  // no config or default value: allow autorotation
+  return UIInterfaceOrientationMaskAllButUpsideDown;
+}
+
+- (UIInterfaceOrientationMask)legacySupportedOrientationsLogic {
```

Comment:
So in my previous comment I wanted to make it clear that `legacySupportedInterfaceOrientations` is a **legacy** version of a method `supportedInterfaceOrientations`. Please keep the original name in such cases.
```suggestion
- (UIInterfaceOrientationMask)legacySupportedInterfaceOrientations
{
```